components/esp_websocket_client/esp_websocket_client.c

```diff
@@ -122,6 +133,52 @@ static uint64_t _tick_get_ms(void)
     return esp_timer_get_time()/1000;
 }
 
+static esp_err_t esp_websocket_setup_rx_buf(esp_websocket_client_handle_t client)
+{
+#ifdef CONFIG_ESP_WS_CLIENT_ENABLE_DYNAMIC_BUFFER
+    if (client->rx_buffer) {
+        free(client->rx_buffer);
+    }
+
+    client->rx_buffer = calloc(1, client->buffer_size);
+    ESP_WS_CLIENT_MEM_CHECK(TAG, client->rx_buffer, return ESP_ERR_NO_MEM);
+#endif
+    return ESP_OK;
+}
+
+static void esp_websocket_free_rx_buf(esp_websocket_client_handle_t client)
```

Comment:
We could merge the `setup` and `free` functions and let the call site to select the buffer:
```cpp
esp_websocket_free_buffer(client->rx_buffer)
```

Reply:
Yeah, we could merge setup and free to one function, but i think seperate them will make sense.

Reply:
I'm not sure if we are talking about the same thing :) . What I meant was to have:
`esp_websocket_free_buffer` and `esp_websocket_setup_buffer`.

Reply:
Yes, i have changed it.